README.md

```diff
@@ -1,48 +1,207 @@
+listed in Section 2.
+
+```ruby
+class Person < ActiveRecord::Base
+  validates :name, presence: true
+end
+
+Person.create(name: "John Doe").valid? # => true
+Person.create(name: nil).valid? # => false
+```
+
+[ar_validations]: http://guides.rubyonrails.org/active_record_validations.html
+
+`#validates()` is our Swiss Army knife for validations. It takes two arguments:
+the first is the name of the attribute we want to validate, and the second is a
+hash of options that will include the details of how to validate it.
+
+In this example, the options hash is `{ presence: true }`, which implements the
+most basic form of validation, preventing the object from being saved if its
+`name` attribute is empty.
+
+# Lifecycle Timing
+
+Before proceeding, keep the answer to this question in mind:
+
+**What is the difference between `#new()` and `#create()`?**
+
+If you've forgotten, `#new()` instantiates a new ActiveRecord model *without*
+saving it to the database, whereas `#create()` immediately attempts to save it,
+as if you had called `#new()` and then `#save()`.
+
+**Database activity triggers validation**. An ActiveRecord model instantiated
+with `#new()` will *not* be validated, because no attempt to write to the
+database was made.
+
+The only way to trigger validation without touching the database is to call the
+`#valid?()` method.
+
+For a full list of methods that trigger validation, see [Section
+4][ar_callbacks_4] of the Rails Guide for Active Record Callbacks. Don't worry
+about the rest of the information in that guide just yet; we'll go into
+callbacks later!
+
+[ar_callbacks_4]: http://guides.rubyonrails.org/active_record_callbacks.html#running-callbacks
+
+# Validation Failure
+
+Here it is, the moment of truth. What can we do when a record fails validation?
+
+## How can you tell when a record fails validation?
+
+**Pay attention to return values!**
+
+By default, ActiveRecord does not raise an exception when validation fails.
+The database operation method will simply return `false` and decline to update
+the database.
+
+Every database method has a sister method with a `!` at the end which will raise
+an exception (`#create!()` instead of `#create()` and so on).
+
+And of course, you can always check manually with `#valid?()`.
+
+## Finding out why validations failed
+
+To find out what went wrong, you can look at the model's `#errors()` object.
+
+You can check all errors at once by examining `errors.messages`.
+
+You can check one attribute at a time by passing the name to `errors` as a key,
+like so:
+
+```ruby
+person.errors[:name]
+```
+
+
+# Displaying Validation Errors in Views
+
+See [Section 8][ar_validations_8] of the Rails Guide for an example of how to
+use the `ActiveModel::Errors#full_messages()` helper, reproduced here for
+convenience:
+
+[ar_validations_8]: http://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views
```

Comment:
I would build up a bit more to this code snippet. Think about the various problems it is solving.

 1. Are there any errors that I need to display?
 2. Yup! Ok how many does the user need to pay attention to?
 3. Ok for each error I need to tell the user. `message` is a bit terse (show them), is there something else? yes! the full_messages array.